sycl/test-e2e/NewOffloadDriver/buffer.cpp

```diff
@@ -0,0 +1,58 @@
+//==------------------- buffer.cpp - SYCL buffer basic test ----------------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+//
+// A basic test using the --offload-new-driver flag.
+
+// REQUIRES: level_zero
+// RUN: %clangxx -fsycl --offload-new-driver %s -o %t.out
+// RUN: %{run} %t.out
+
+#include <sycl/detail/core.hpp>
+
+int main() {
+  // Creating buffer of 4 elements to be used inside the kernel code.
+  sycl::buffer<size_t, 1> Buffer(4);
+
+  // Creating SYCL queue.
+  sycl::queue Queue;
```

Comment:
probably we don't want to add this file as part of this pr, looks good otherwise tho, thx.